hazelcast/src/main/java/com/hazelcast/jet/impl/JetClientInstanceImpl.java

```diff
@@ -126,6 +143,83 @@ public Job newJobProxy(long jobId, boolean isLightJob, @Nonnull Object jobDefini
         return new ClientJobProxy(client, jobId, isLightJob, jobDefinition, config);
     }
 
+    @Override
+    public void submitJobFromJar(@Nonnull SubmitJobParameters submitJobParameters) {
+
+        try {
+            UUID sessionId = UuidUtil.newSecureUUID();
+
+            Path jarPath = submitJobParameters.getJarPath();
+            // Calculate some parameters for JobMetadata
+            String fileNameWithoutExtension = getFileNameWithoutExtension(jarPath);
+            String sha256Hex = calculateSha256Hex(jarPath);
+            long jarSize = Files.size(jarPath);
+
+            // Send job meta data
+            boolean result = sendJobMetaData(sessionId, fileNameWithoutExtension, sha256Hex, submitJobParameters);
+            if (result) {
+                logFine(getLogger(), "Submitted JobMetaData successfully for jarPath: %s", jarPath);
+            }
```

Comment:
Is there a reason to process job parts if metadata submission was not successful?

Reply:
There is no reason. If metadata upload fails, exception is thrown and upload is aborted. But we still log the result of the calls if it is successful